dask/dataframe/io/parquet.py

```diff
@@ -939,15 +941,39 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
     --------
     to_parquet
     """
-    read = get_engine(engine)['read']
-
-    fs, fs_token, paths = get_fs_token_paths(path, mode='rb',
-                                             storage_options=storage_options)
-
-    if isinstance(path, string_types) and len(paths) > 1:
-        # Sort paths naturally if multiple paths resulted from a single
-        # specification (by '*' globbing)
-        paths = sorted(paths, key=natural_sort_key)
+    is_ParquetFile = False
+    try:
+        import fastparquet
+        if isinstance(path, fastparquet.api.ParquetFile):
+            if path.open != fastparquet.util.default_open:
+                assert (re.match('.*://', path.fn)), \
+                       ("ParquetFile: Path must contain protocol" +
+                        " (e.g., s3://...) when using other than the default" + 
+                        " LocalFileSystem. Path given: " + path.fn)
+
+            assert (path.file_scheme == 'hive'), "Only 'hive' is currently supported as the ParquetFile's file scheme"
```

Comment:
Did you check for non-hive? I feel like it ought to work fine. If not, the message could be a bit clearer: only hive is supported when passing a fastparquet.ParquetFile instance.

Reply:
You are right. It works with non-hive. I made that assumption before being very familiar with it. Code is now supporting non-hive.
This PR was already useful for me to learn more about this stuff.